index.js

```diff
@@ -1,36 +1,16 @@
 const fromIter = iter => (start, sink) => {
   if (start !== 0) return;
-  const iterator =
-    typeof Symbol !== 'undefined' && iter[Symbol.iterator]
-      ? iter[Symbol.iterator]()
-      : iter;
-  let inloop = false;
-  let got1 = false;
-  let completed = false;
-  let res;
-  function loop() {
-    inloop = true;
-    while (got1 && !completed) {
-      got1 = false;
-      res = iterator.next();
-      if (res.done) {
-        sink(2);
-        break;
-      }
-      else sink(1, res.value);
-    }
-    inloop = false;
-  }
+  const isIterable = typeof Symbol !== 'undefined' && iter[Symbol.iterator];
+  const iterator = isIterable ? iter[Symbol.iterator]() : iter;
+  let disposed = false;
+  let result = {};
   sink(0, t => {
-    if (completed) return
-
-    if (t === 1) {
-      got1 = true;
-      if (!inloop && !(res && res.done)) loop();
-    } else if (t === 2) {
-      completed = true;
-    }
+    if (disposed) return;
+    if (t === 1) result = iterator.next();
+    if (t === 2) disposed = true;
   });
+  while (!result.done && !disposed) sink(1, result.value);
```

Comment:
This might cause infinie loop. U assume in this implementation that u always receive pull immediately in response to pushing the value down

Reply:
this example works normally. What you mentioned can occur just if for some reason a sink calls the source with 0 and doesn't immediately talkback with 1 or 2. That's true. I just wonder if that's a real scenario, I mean forEach, iterate, subscribe, all of them call `source(0, (t, d) => { })` and in case of `t === 0 || t === 1` there is an immediate `talkback(1)`. Is that in compliance with the spec? I believe delays should happen through operators and not in the sink by not calling tb(1).

Reply:
Implemented sinks (like forEach, subscribe etc) indeed pull automatically - but this is not something that spec assumes. You could also implement `backpressurePulls` operator that could delay pulls sent upwards and this implementation wouldnt handle it well.

Also from what I see this implementation doesnt handle my delay example (I havent tested it though and I might be wrong). I think it will hang in the while loop, because it will try to send down the same value, rather than next ones.

Reply:
I tested it, it works fine if working with delay operators without patching the sinks we know.
In any case, I believe the original version that is implemented by Andre now should include these test cases you mentioned in test.js

Reply:
I'm gonna clone the PR because it really doesnt support that a lazy sink that talks back after a delay, thanks @Andarist for the keen eye